### PR TITLE
Add `native_dio_client` to readme

### PR DESCRIPTION
**Pull request by ueman**
### New Pull Request Checklist

- [x] I have read the [Documentation](https://pub.dartlang.org/packages/dio)
- [x] I have searched for a similar pull request in the [project](https://github.com/flutterchina/dio/pulls) and found none
- [x] I have updated this branch with the latest `develop` to avoid conflicts (via merge from master or rebase)
- [x] I have added the required tests to prove the fix/feature I am adding
- [x] I have updated the documentation (if necessary)
- [x] I have run the tests and they pass

### Pull Request Description

This adds [`native_dio_client`](https://github.com/ueman/native-dio-client) to the readme.

### CODE REVIEW

File: README-ZH.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ void getHttp() async {
 | [dio_cache_interceptor](https://github.com/llfbandit/dio_cache_interceptor) | [![Pub](https://img.shields.io/pub/v/dio_cache_interceptor.svg?style=flat-square)](https://pub.dev/packages/dio_cache_interceptor) | Dio HTTP cache interceptor with multiple stores respecting HTTP directives (or not)             |
 | [dio_http_cache](https://github.com/hurshi/dio-http-cache) | [![Pub](https://img.shields.io/pub/v/dio_http_cache.svg?style=flat-square)](https://pub.dev/packages/dio_http_cache) | A simple cache library for Dio like Rxcache in Android             |
 | [pretty_dio_logger](https://github.com/Milad-Akarie/pretty_dio_logger) | [![Pub](https://img.shields.io/pub/v/pretty_dio_logger.svg?style=flat-square)](https://pub.dev/packages/pretty_dio_logger) | Pretty Dio logger is a Dio interceptor that logs network calls in a pretty, easy to read format.            |
+| [native_dio_client](https://github.com/ueman/native-dio-client) | [![Pub](https://img.shields.io/pub/v/native_dio_client.svg?style=flat-square)](https://pub.dev/packages/native_dio_client) | An adapter for Dio which makes use of cupertino_http and cronet_http to delegate HTTP requests to the native platform. |
```
Comment (AlexV525) [inline README-ZH.md:50]:
```suggestion
| [native_dio_client](https://github.com/ueman/native-dio-client) | [![Pub](https://img.shields.io/pub/v/native_dio_client.svg?style=flat-square)](https://pub.dev/packages/native_dio_client) | 使用 cupertino_http 和 cronet_http 以 Dio 适配器代理实现的原生网络请求功能。 |
```

**Review by AlexV525 — COMMENTED** [review]
LGTM. However, I saw previous descriptions are not in Chinese so maybe it can start from this?

**Comment by ueman** [comment]
@AlexV525

The build failures arent related to the proposed changes since this is a doc only change.

I would propose that you or any of the other maintainers (@kuhnroyal?) fix those, so that not every PR has to make those changes individually.

**Comment by AlexV525** [comment]
> I would propose that you or any of the other maintainers (@kuhnroyal?) fix those, so that not every PR has to make those changes individually.

It's actually fixed by #1539. I don't think maintainers of this repo are still contributing actively. We can leave CIs as is.

**Pull request closed without merging**